source/blog/2018-11-19-ember-3-6-released.md

```diff
@@ -0,0 +1,207 @@
+
+* `new` syntax is not currently supported with classes that extend from `EmberObject`. You must continue to use the `create` method when making new instances of classes, even if they are defined using native class syntax. If you want to use `new` syntax, consider creating classes which do _not_ extend from `EmberObject`. Ember features, such as computed properties and decorators, will still work with base-less classes.
+* Instead of using `this._super()`, you should must standard `super` syntax. See the [MDN docs on classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Super_class_calls_with_super) for more details.
+* Native classes support using [constructors](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Constructor) to set up newly-created instances. Ember uses these to, among other things, support features that need to retrieve other entities by name, like Service injection and `getOwner`. To ensure your custom instance setup logic takes place after this important work is done, avoid using the `constructor` in favor of `init`.
+* Using native classes, and switching back to the old Ember Object model is fully supported.
+* For early adopters who are used to argument values and values passed to `create` being overriden, this is no longer the case! Class field values will be the default, and any value passed to a class on creation will override that value.
+
+**Compatibility and Polyfill**
+
+A polyfill for this behavior has been built which backports this behavior to both Ember 3.4 and 3.5. You can see it [here](https://github.com/pzuraq/ember-native-class-polyfill), or install it using ember-cli:
+
+```sh
+ember install ember-native-class-polyfill
+```
+
+We would like to add support for prior LTS versions (2.18 and 2.16) as well, if you would like to contribute, ping us in the #st-native-classes channel on Discord or in the [Native Class Quest issue](https://github.com/emberjs/ember.js/issues/16927) on Github!
+
+**Final stage of the router service RFC (2 of 2)**
+
+<!--alex ignore rejects -->
+Ember 3.6 introduces the final stages of the router service RFC. The includes two new methods: `recognize(url)` that can return a `RouteInfo` based on the URL you pass and `recognizeAndLoad(url)` that takes a string URL and returns a promise that resolves to a `RouteInfoWithAttributes` for the leaf-most route represented by the URL. The promise rejects if the URL is not recognized or an unhandled exception is encountered.
+
+This final stage also introduces the new observable property `currentRoute`. It is guaranteed to change whenever a route transition happens (even when that transition only changes parameters and doesn't change the active route). You should consider its value deeply immutable -- we will replace the whole structure whenever it changes. The value of `currentRoute` is a `RouteInfo` representing the current leaf route.
```

Comment:
What is meant by deeply immutable?

Reply:
It means that it is read-only which I now realise might be a better way to teach it.

Reply:
It sounds like not only is the value read-only, but that no one should try to read the value at all. Is that accurate?

We could say something like "While the property may be observed for changes, its value should be neither read nor modified. The values are Private API, meaning that the structure of the value may be changed in breaking ways without warning."

Reply:
We need to document this very very clearly in the API docs too. I'm taking a look to see what they say.

Reply:
https://github.com/emberjs/ember.js/pull/17214

Before:

>      A RouteInfo that represents the current leaf route.
>      It is guaranteed to change whenever a route transition
>      happens (even when that transition only changes parameters
>      and doesn't change the active route)

After

>      ... and doesn't change the active route).
>      This property may be observed for changes, as public API.
>      However, its _value_ is private API, and should not be depended on.

Reply:
Hmm, I'm honestly a bit confused about this. I think you're right in that it can be observed for changes but the values can't be modified. Still a bit uncertain about the ability to read. Hopefully, they can provide a clear answer. Thanks for catching this!

Reply:
Confirmed with @chadhietala that this is intended to be public API (including the object's themselves).  More details are included in the [Router Service RFC](https://github.com/emberjs/rfcs/blob/master/text/0095-router-service.md).

Reply:
Thanks!